enterprise_gateway/services/kernels/remotemanager.py

```diff
@@ -423,3 +455,24 @@ def _get_process_proxy(self):
                        format(self.kernel_spec.display_name, process_proxy_class_name))
         process_proxy_class = import_item(process_proxy_class_name)
         self.process_proxy = process_proxy_class(kernel_manager=self, proxy_config=process_proxy_cfg.get('config'))
+
+    @property
```

Comment:
I think we should add a comment block somewhere (thinking prior to these properties) that talks about when `self.parent` would not exist.  This might warrant an entry in the docs as well - although I don't mind keeping this on the down-low for the time being. :wink:

Reply:
Will do. Is there somewhere specific in the docs where something like this might go?

Reply:
I'm okay deferring this for now, although if we were to add it, the following locations might be worthy of updates:
- We should add an entry (or two?) to [Use Cases](https://jupyter-enterprise-gateway.readthedocs.io/en/latest/use-cases.html#use-cases)
- Perhaps add the details to the end of the deployments section (following [IBM Spectrum Conductor](https://jupyter-enterprise-gateway.readthedocs.io/en/latest/kernel-conductor.html#ibm-spectrum-conductor))

I'm trying to think of what we'd call the topic.  The term "Library Mode" popped into my head just now.  I'm not enamored with that, but I'd like the topic to convey a more direct capability that utilizes the heart of EG.  "Integration Mode" - something that allows third-parties to use RMK directly.  Ideas welcome! 

The topic should include an example - much like the one you opened the issue with - based on the final results.  (Btw, I'd like to have those details anyway so I could try things out against my Hadoop YARN cluster.)

Reply:
I like library mode. Just anything that conveys the meaning that this is JEG but without the gateway webapp.

Standalone Remote Kernel Execution if verbosity isn't an issue. 

The example actually becomes really simple.

```
import nbformat
from nbclient import NotebookClient
from enterprise_gateway.services.kernels.remotemanager import RemoteKernelManager

with open("/home/ish/notebooks/Untitled.ipynb") as fp:
    test_notebook = nbformat.read(fp, as_version=4)

client = NotebookClient(nb=test_notebook, kernel_manager_class=RemoteKernelManager)
client.execute()
```

If you'd like a full docker with kernelspecs and config, I can do that, but I don't have experience with Yarn so it may take a while.

Reply:
I like "Standalone Remote Kernel Execution" as the topic.  Perhaps in the opening statement, we can toss in an "a.k.a., Library Mode". :smile: This way we have a way to quickly identify the context when communicating this functionality or discussing it in general.

The sample is excellent - I'm really happy with how this is turning out.  Let's not worry about a docker image for the moment. Given the example, I can run this against my YARN cluster, although probably not today (sorry).